Notebooks/transcript_dataframe_builder.ipynb

```diff
@@ -0,0 +1,1264 @@
+    "import pandas as pd\n",
+    "import os "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Connect to Council's Data Base"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## For Seattle\n",
+    "fireo.connection(client=Client(\n",
+    "    project=\"cdp-seattle-staging-dbengvtn\",\n",
+    "    credentials=AnonymousCredentials()\n",
+    "))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Connect to Council's File Store"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## For Seattle\n",
+    "fs = GCSFileSystem(project=\"cdp-seattle-staging-dbengvtn\", token=\"anon\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## Fetch details for all available transcripts for council of interest\n",
+    "transcript_model = list(db_models.Transcript.collection.fetch())[0:]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "58"
+      ]
+     },
+     "execution_count": 14,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "## Total transcripts for this particular council at this time\n",
+    "len(transcript_model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Dowload all transcripts to local machine\n",
+    "__Note:__ _The function below is for downloading these transcripts to your local machine and therefore may be time and space intensive._"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## Finish documenting function\n",
+    "\n",
+    "## Function to Download all transcripts locally\n",
+    "def download_transcripts(list_of_transcripts_db_ref, council):\n",
+    "    '''\n",
+    "    Takes a list of database objects, and a chosen council as arguments.\n",
+    "    council argument must be in string format.\n",
+    "    Creates a file for transcripts to download to.\n",
+    "    Downloads database objects to local machine as .json type.\n",
+    "    Transcript files are saved with name incrementally counting up from 0.\n",
+    "    '''\n",
+    "    num = 0  ## For adding index to file names\n",
+    "    for transcripts in range(0, len(list_of_transcripts_db_ref)):\n",
+    "        s = str(num)\n",
+    "        fs.get(transcript_model[0].file_ref.get().uri, \n",
+    "               \"transcripts/\"+council+\"/\"+council+\"-transcript\"+s+\".json\")\n",
+    "        del transcript_model[0]\n",
+    "        num = num + 1\n",
+    "    print(\"Transcript download complete!\")"
```

Comment:
I have thoughts about this function :joy: 

We could technically write a function that loads transcripts just-in-time for compute. Rather than storing on user machine... We don't have enough data for that to matter right now.

At the very least we could multithread this / make a function in a library for downloading the dataset.

Reply:
I agree, to be honest I was trying to do this all in place one night and got stuck...for a looong time lol.

So as a quick fix so I could get cleaning and processing data for EDA I just downloaded everything.

I do plan on changing this